### PR TITLE
Feature/te 336 release job

### PR DESCRIPTION
**Pull request by franzbecker**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+    stage 'Build target platform'
+    withMavenEnv {
+        mvn 'clean install -f "releng/org.testeditor.releng.target/pom.xml"'
+    }
+
+    stage (isMaster() ? 'Build and deploy' : 'Build')
+    withMavenEnv(["MAVEN_OPTS=-Xms512m -Xmx2g"]) {
+        // workaround for now to speed-up the build: only build the product on develop and master
+        def buildProduct = env.BRANCH_NAME == 'develop' || isMaster() ? '-Pproduct' : ''
```
Comment (gunther-bachmann) [inline Jenkinsfile]: how about isDevelop()

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        sh "git checkout develop"
+        sh "git reset --hard origin/develop"
+        def developVersion = getCurrentVersion()
+        if (developVersion == preReleaseVersion) {
+            sh "git merge origin/master"
+            def nextSnapshotVersion = '\\${parsedVersion.majorVersion}.\\${parsedVersion.minorVersion}.\\${parsedVersion.nextIncrementalVersion}-SNAPSHOT'
+            setVersion(nextSnapshotVersion, 'releng/org.testeditor.releng.target/pom.xml', 'org.testeditor.releng.target.parent')
+            setVersion(nextSnapshotVersion, 'pom.xml', 'org.testeditor.releng.parent')
+            sh "git add *"
+            sh "git commit -m '[release] set version ${getCurrentVersion()}'"
```
Comment (gunther-bachmann) [inline Jenkinsfile]: should be nextSnapshotVersion, right?

File: Jenkinsfile (hunk 1 above)
Comment (franzbecker) [inline Jenkinsfile]: We only need this here, would introduce it as soon as we've got another occurrence.

File: Jenkinsfile (hunk 2 above)
Comment (franzbecker) [inline Jenkinsfile]: The closure `${getCurrentVersion()}` retrieves the current version after the `setVersion` was executed. `nextSnapshotVersion` is only a pattern that is passed to Maven and gets interpreted. The script doesn't now the next version yet, so it needs to read it from the POM.

**Pull request merged**